drivers/SmartThings/zigbee-window-treatment/src/screen-innovations/init.lua

```diff
@@ -0,0 +1,174 @@
+-- require st provided libraries
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local window_preset_defaults = require "st.zigbee.defaults.windowShadePreset_defaults"
+local device_management = require "st.zigbee.device_management"
+local utils = require "st.utils"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+local PowerConfiguration = clusters.PowerConfiguration
+
+-- manufacturer specific cluster details
+local CUS_CLU = 0xFCCC
+local RUN_DIR_ATTR = 0x0012
+
+
+local motor_states = {
+  IDLE = 0,
+  OPENING = 1,
+  CLOSING = 2
+}
+local running_direction = motor_states.IDLE
```

Comment:
Variables declared this way are in scope for _all_ devices on the driver, meaning that if you have two window shades, they'll both be writing and reading to the same value. 

You'll need to use `device:set_field(...)` to use the device's local store instead.

Reply:
Hi @greens, 
Updated to maintain device specific states.